crates/core/src/client/message_handlers.rs

```diff
@@ -136,19 +142,36 @@ enum DecodedMessage<'a> {
 }
 
 impl DecodedMessage<'_> {
-    async fn handle(self, client: &ClientConnection) -> Result<(), MessageExecutionError> {
+    async fn handle(self, client: &ClientConnection, timer: Instant) -> Result<(), MessageExecutionError> {
+        let address = client.module.info().address;
         let res = match self {
             DecodedMessage::Call { reducer, args } => {
                 let res = client.call_reducer(reducer, args).await;
+                DB_METRICS
+                    .request_round_trip
+                    .with_label_values(&WorkloadType::Reducer, &address, reducer)
+                    .observe(timer.elapsed().as_secs_f64());
                 res.map(drop).map_err(|e| (Some(reducer), e.into()))
             }
             DecodedMessage::Subscribe(subscription) => {
-                client.subscribe(subscription).await.map_err(|e| (None, e.into()))
+                let res = client.subscribe(subscription).await;
```

Comment:
Similarly when does this return? After the initial subscription has been processed or before that? And could we add a comment to that effect?

Reply:
we do have await all the way down till initial subscription proccessed. I will add a comment